src/OrleansProviders/Streams/Common/SimpleCache/SimpleQueueCache.cs

```diff
@@ -244,40 +242,24 @@ internal bool TryGetNextMessage(SimpleQueueCacheCursor cursor, out IBatchContain
             Log(logger, "TryGetNextMessage: {0}", cursor);
 
             batch = null;
+            if (!cursor.IsSet) return false;
 
-            if (cursor == null) throw new ArgumentNullException("cursor");
-            
-            //if not set, try to set and then get next
-            if (!cursor.IsSet)
-            {
-                InitializeCursor(cursor, cursor.SequenceToken, false);
-                return cursor.IsSet && TryGetNextMessage(cursor, out batch);
-            }
-
-            // has this message been purged
-            if (cursor.SequenceToken.Older(cachedMessages.Last.Value.SequenceToken))
-            {
-                throw new QueueCacheMissException(cursor.SequenceToken, cachedMessages.Last.Value.SequenceToken, cachedMessages.First.Value.SequenceToken);
-            }
-
-            // Cursor now points to a valid message in the cache. Get it!
             // Capture the current element and advance to the next one.
             batch = cursor.Element.Value.Batch;
-            
-            // Advance to next:
+
+            // If we are at the end of the cache unset cursor and move offset one forward
             if (cursor.Element == cachedMessages.First)
             {
-                // If we are at the end of the cache unset cursor and move offset one forward
-                ResetCursor(cursor, ((EventSequenceToken)cursor.SequenceToken).NextSequenceNumber());
+                UnsetCursor(cursor, null);
```

Comment:
Should this be null? It seems like the code accommodates for a null sequence token, treating it as a default, but I'm not 100% sure.